shared/data/fingerprint-protection.js

```diff
@@ -291,6 +280,32 @@
         return windowScript
     }
 
+    /**
+     * Temporary storage can be used to determine hard disk usage and size.
+     * This will limit the max storage to 4GB without completely disabling the
+     * feature.
+     */
+    function modifyTemporaryStorage () {
+        const randomFunctionName = 'A' + Math.random().toString(36).substring(2) // random string of alphanumeric chars
+        const script = `
+            if (navigator.webkitTemporaryStorage) {
+                try {
+                    navigator.webkitTemporaryStorage.${randomFunctionName} = navigator.webkitTemporaryStorage.queryUsageAndQuota
```

Comment:
hm, why are we storing it as a property and not a variable in this scope? Sth like:

```js
                    const org = navigator.webkitTemporaryStorage.queryUsageAndQuota
///…
                    org.call(navigator.webkitTemporaryStorage, modifiedCallback, err)
```

Reply:
Good catch Konrad, should have used call from the start, updated.